packages/@winglang/wingc/src/jsify.rs

```diff
@@ -2146,6 +2186,397 @@ impl<'a> JSifier<'a> {
+			TypeAnnotationKind::Inferred => panic!("Unexpected inferred type annotation"),
+			TypeAnnotationKind::Void => new_code!(expr_span, "std.Type._ofVoid()"),
+			TypeAnnotationKind::Optional(t) => new_code!(
+				expr_span,
+				"std.Type._ofOptional(",
+				self.jsify_reflection_udt(&t, &t.span, ctx),
+				")"
+			),
+			TypeAnnotationKind::Array(t) => new_code!(
+				expr_span,
+				"std.Type._ofArray(",
+				self.jsify_reflection_udt(&t, &t.span, ctx),
+				", false)"
+			),
+			TypeAnnotationKind::MutArray(t) => new_code!(
+				expr_span,
+				"std.Type._ofArray(",
+				self.jsify_reflection_udt(&t, &t.span, ctx),
+				", true)"
+			),
+			TypeAnnotationKind::Map(t) => new_code!(
+				expr_span,
+				"std.Type._ofMap(",
+				self.jsify_reflection_udt(&t, &t.span, ctx),
+				", false)"
+			),
+			TypeAnnotationKind::MutMap(t) => new_code!(
+				expr_span,
+				"std.Type._ofMap(",
+				self.jsify_reflection_udt(&t, &t.span, ctx),
+				", true)"
+			),
+			TypeAnnotationKind::Set(t) => new_code!(
+				expr_span,
+				"std.Type._ofSet(",
+				self.jsify_reflection_udt(&t, &t.span, ctx),
+				", false)"
+			),
+			TypeAnnotationKind::MutSet(t) => new_code!(
+				expr_span,
+				"std.Type._ofSet(",
+				self.jsify_reflection_udt(&t, &t.span, ctx),
+				", true)"
+			),
+			TypeAnnotationKind::Function(t) => {
+				let mut func_code = new_code!(expr_span, "std.Type._ofFunction(new std.FunctionType(");
+				func_code.append(match t.phase {
+					Phase::Inflight => "std.Phase.INFLIGHT",
+					Phase::Preflight => "std.Phase.PREFLIGHT",
+					Phase::Independent => "std.Phase.UNPHASED",
+				});
+				func_code.append(", [");
+				for p in &t.parameters {
+					func_code.append(self.jsify_reflection_udt(&p.type_annotation, &p.type_annotation.span, ctx));
+					func_code.append(", ");
+				}
+				func_code.append("], ");
+				func_code.append(self.jsify_reflection_udt(&t.return_type, &t.return_type.span, ctx));
+				func_code.append("))");
+				func_code
+			}
+			TypeAnnotationKind::UserDefined(udt) => {
+				let type_ = resolve_user_defined_type_ref(
+					udt,
+					ctx.visit_ctx.current_env().expect("no current env"),
+					ctx.visit_ctx.current_stmt_idx(),
+				)
+				.expect("type reference");
+
+				self.jsify_reflection_type(*type_, expr_span)
+			}
+		}
+	}
+
+	fn jsify_reflection_type(&self, type_: TypeRef, expr_span: &WingSpan) -> CodeMaker {
+		let type_reflection_definitions = self.type_reflection_definitions.borrow();
+		if let Some((variable_name, _, _)) = type_reflection_definitions.get(&type_) {
+			return new_code!(expr_span, format!("$types.{variable_name}"));
+		}
+		std::mem::drop(type_reflection_definitions);
+
+		// Generate a unique name for referring to this type in the $types object
+		let mut type_variable_counter = self.type_variable_counter.borrow_mut();
+		*type_variable_counter += 1;
+		let type_variable_name = format!("t{}", type_variable_counter);
+		std::mem::drop(type_variable_counter);
```

Comment:
Give the variables short friendly names before the unique number to make them more debuggable